File: 프로그래머스/lv2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        Map<String, Integer> map = new HashMap<>();
        
        for (String[] c : clothes) {
            map.put(c[1], map.getOrDefault(c[1], 0) + 1);
        }
        
        int answer = 1;
        for (String s : map.keySet()) {
            answer *= (map.get(s) + 1);
        }
        
        return answer - 1;
    }
}

File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;
map<string,int> ma;
int solution(vector<vector<string>> clothes) {
    int answer = 1;
    for(int i=0;i<clothes.size();i++){
        ma[clothes[i][1]]++;
    }
    for(auto&i:ma){
        answer*=(i.second+1);
    }
    return answer-1;
}